sshconnect2.c

```diff
@@ -1262,10 +1262,20 @@ identity_sign(struct identity *id, u_char **sigp, size_t *lenp,
 			goto out;
 		}
 		sign_key = prv;
-		if (sshkey_is_sk(sign_key)) {
+		if (sshkey_is_sk(sign_key) &&
+		    (sign_key->sk_flags & SSH_SK_USER_PRESENCE_REQD)) {
```

Comment:
this block goes on to check for SSH_SK_USER_VERIFICATION_REQD keys. Is it possible to have a key that wants UV but not UP? If so, I think it will skip prompting for PIN

Reply:
Sorry, but I don't understand your comment. This block does not check for "uv" at all. It just notices the user that the user *may* have to touch the authenticator, because at this point we don't know, and the presence check may be required right in the first call to sshkey_sign, if "uv" isn't required.